flexget/plugins/filter/charts.py

```diff
@@ -0,0 +1,258 @@
+
+
+@db_schema.upgrade('charts_snep')
+def upgrade(ver, session):
+    return 0
+
+
+class ChartsProvider(Base):
+    __tablename__ = 'charts_provider'
+    id = Column(Integer, primary_key=True)
+    organization = Column(String, nullable=False) # Company, association, person etc who provides charts
+    category = Column(String) # radio, album, streaming, all etc
+    periodicity = Column(String)
+    plugin_name = Column(String)
+    releases = relation('ChartsRelease',
+                        backref='provider',
+                        cascade='all, delete, delete-orphan')
+
+
+class ChartsRelease(Base):
+    __tablename__ = 'charts_release'
+    id = Column(Integer, primary_key=True)
+    provider_id = Column(Integer, ForeignKey('charts_provider.id'), nullable=False)
+    expires = Column(DateTime, nullable=False)
+    entries = relation('ChartsEntry',
+                       backref='release',
+                       cascade='all, delete, delete-orphan')
+
+
+class ChartsEntry(Base):
+    __tablename__ = 'charts_entry'
+    id = Column(Integer, primary_key=True)
+    release_id = Column(Integer, ForeignKey('charts_release.id'), nullable=False)
+
+    # TODO: Is enough generic?
+    rank = Column(Integer)
+    best_rank = Column(Integer)
+    charted_weeks = Column(Integer)
+    artist = Column(Unicode)
+    title = Column(Unicode)
+# endregion
+
+
+class ChartsPlugin(object):
+    """
+    Example:
+    charts:
+      provider : snep
+      category : radio
+      max_best_rank : 5
+      min_charted_weeks : 4
+    """
+    schema = {
+        'type': 'object',
+        'properties': {
+            'provider': {'type': 'string'},
+            'category': {'type': 'string'},
+            'max_rank': {'type': 'integer'},
+            'max_best_rank': {'type': 'integer'},
+            'min_charted_weeks': {'type': 'integer'}
+        },
+        'additionalProperties': False,
+        'required': ['provider', 'category']
+    }
+
+    def get_last_release_online(self, organization, category=None, periodicity=None):
+        """
+        Retrieves online the last charts release for a given provider.
+        :rtype ChartsRelease
+        """
+        connector = charts_connectors.get(organization)
+        if connector is None:
+            log.warn("Charts connector for '%s' not found!" % organization)
+            return None
+
+        return connector.retrieve_charts(charts_type=category, date_interval=periodicity)
+
+    def get_provider_id(self, organization, category=None, periodicity=None, auto_create=False):
+        """
+        Return the provider id for the given filters. If auto_create is set to True
+        and the provider doesn't exist in DB then this method will create one and
+         return its id.
+        :type organization: str
+        :type category: str
+        :type periodicity: str
+        :type auto_create: bool
+        :rtype: int
+        """
+
+        session = Session()
```

Comment:
I think it's preferable to use session as a context manager, more readable IMO:
```python
with Session as session:
```
That way you could ditch the try, finally method which works fine, but kinda hacky for this purpose.
Also you can use the `@with_session` decorator